lib/vector-buffers/src/variants/disk_v2/v1_migration.rs

```diff
@@ -73,31 +72,34 @@ where
     info!("Detected old `disk_v1`-based buffer for the `{}` sink. Automatically migrating to `disk_v2`.", id);
 
     let mut migrated_records = 0;
-    while let Some(old_record) = src_reader.next().await {
+    while let Some(mut old_record) = src_reader.next().await {
         let old_record_event_count = old_record.event_count();
+        let finalizers = ManuallyDrop::new(old_record.take_finalizers());
 
         dst_writer.write_record(old_record).await.map_err(|e| {
+            finalizers.update_status(EventStatus::Errored);
             format!(
                 "failed writing record {} to the new disk v2 buffer: {}",
                 migrated_records, e,
             )
         })?;
 
         dst_writer.flush().await.map_err(|e| {
+            finalizers.update_status(EventStatus::Errored);
```

Comment:
Do we actually need to update the finalizer status if they're never going to be pushed in the first place? Given that we don't drop them unless we have a successful write/flush?

Reply:
That is extraneous for now. I had put in the status update to start easing the changes for when the buffers handle nacks. I can go either way on it since this block will need to be modified (to remove the `ManuallyDrop`) in either case.

Reply:
Ah, gotcha. Works for me. 👍🏻